server/aws-lsp-codewhisperer/src/language-server/chat/contexts/documentContext.test.ts

```diff
@@ -0,0 +1,163 @@
+import { EditorState } from '@amzn/codewhisperer-streaming'
+import * as assert from 'assert'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { DocumentContextExtractor } from './documentContext'
+
+describe('DocumentContext', () => {
+    const mockTypescriptCodeBlock = `function test() {
+    console.log('test')
+}`
+    const mockTSDocument = TextDocument.create('file://test.ts', 'typescript', 1, mockTypescriptCodeBlock)
+
+    describe('documentContextExtractor.extractEditorState', () => {
+        it('extracts editor state for range selection', async () => {
+            const documentContextExtractor = new DocumentContextExtractor(19)
+            const expected: EditorState = {
+                document: {
+                    programmingLanguage: { languageName: 'typescript' },
+                    relativeFilePath: 'file://test.ts',
+                    documentSymbols: [],
+                    text: "console.log('test')",
+                },
+                cursorState: {
+                    range: {
+                        start: {
+                            line: 0,
+                            character: 8,
+                        },
+                        end: {
+                            line: 0,
+                            character: 11,
+                        },
+                    },
+                },
+            }
+
+            const result = await documentContextExtractor.extractEditorState(mockTSDocument, {
+                // highlighing "log"
+                range: {
+                    start: {
+                        line: 1,
+                        character: 12,
+                    },
+                    end: {
+                        line: 1,
+                        character: 15,
+                    },
+                },
+            })
+
+            assert.deepStrictEqual(result, expected)
+        })
+
+        it('extracts editor state for collapsed poisition', async () => {
```

Comment:
nit: typo

Reply:
I don't understand why this test returns line 2, when the cursor state is on line 1?

Reply:
Nevermind, the result is transposed by taking the limit characters, so the lines that are cut off are also cut off from the returned `cursorState` to make it match.